src/rgw/driver/rados/rgw_bucketlogging.cc

```diff
@@ -0,0 +1,605 @@
+      ldpp_dout(this, 10) << "ERROR: failed to load bucket : "
+                          << bucket_name << " , error: " << ret << dendl;
+      return ret;
+    }
+    ldpp_dout(this, 10) << "INFO: loaded bucket : " << bucket_name
+                        << ", id: "<< bucket_id << dendl;
+    ret = log_bucket->commit_logging_object(entry, yield, this, prefix, nullptr, false);
+    if (ret < 0) {
+      ldpp_dout(this, 10) << "ERROR: failed to commit logging object : "
+                          << entry << " , error: " << ret << dendl;
+    }
+    return ret;
+  }
+
+  void async_sleep(boost::asio::yield_context yield, const std::chrono::milliseconds& duration) {
+    using Clock = ceph::coarse_mono_clock;
+    using Timer = boost::asio::basic_waitable_timer<Clock,
+        boost::asio::wait_traits<Clock>, Executor>;
+    Timer timer(m_io_context);
+    timer.expires_after(duration);
+    boost::system::error_code ec;
+    timer.async_wait(yield[ec]);
+    if (ec) {
+      ldpp_dout(this, 10) << "ERROR: async_sleep failed with error: "
+                         << ec.message() << dendl;
+    }
+  }
+
+  // unlock (lose ownership) list object
+  int unlock_list_object(const std::string& list_obj_name, boost::asio::yield_context yield) {
+
+    librados::ObjectWriteOperation op;
+    op.assert_exists();
+    rados::cls::lock::unlock(&op, list_obj_name+"_lock", m_lock_cookie);
+    auto& rados_ioctx = m_rados_store->getRados()->get_logging_pool_ctx();
+    const auto ret = rgw_rados_operate(this, rados_ioctx, list_obj_name,
+                                       std::move(op), yield);
+    if (ret == -ENOENT) {
+      ldpp_dout(this, 10) << "INFO: log commit list: " << list_obj_name
+        << ". was removed. nothing to unlock" << dendl;
+      return 0;
+    }
+    if (ret == -EBUSY) {
+      ldpp_dout(this, 10) << "INFO: log commit list: " << list_obj_name
+        << ". already owned by another RGW. no need to unlock" << dendl;
+      return 0;
+    }
+    return ret;
+  }
+
+  // processing of a specific target list
+  void process_commit_list(const std::string& list_obj_name, boost::asio::yield_context yield) {
+  //  constexpr auto max_elements = 1024;
+    auto is_idle = false;
+    const std::string start_marker;
+
+    std::string tenant_name, bucket_name, bucket_id, prefix;
+    int ret = parse_list_name(list_obj_name, tenant_name, bucket_name,
+                              bucket_id, prefix);
+    if (ret) {
+      ldpp_dout(this, 10) << "ERROR: list name: " << list_obj_name
+                          << " is invalid. Processing will stop" << dendl;
+      return;
+    }
+    ldpp_dout(this, 20) << "INFO: parse list_name: " << list_obj_name
+                        << ", bucket_name: " << bucket_name
+                        << ", bucket_id: " << bucket_id
+                        << ", tenant_name: " << tenant_name
+                        << ", prefix: " << prefix << dendl;
+
+    // Sanity check : can the bucket be loaded?
```

Comment:
not sure that we need that here?
or maybe move it inside the while loop, fo the case that the bucker is deleted but there are not pending commits, so we would never fail committing to the bucket?